Cargo.toml

```diff
@@ -35,6 +37,7 @@ std = ["alloc"]
 trace-pkt = ["alloc"]
 paranoid_unsafe = []
 core_error = []
+sync = ["maybe-async/is_sync"]
```

Comment:
Whether or not `gdbstub` should present a sync vs. async API by default is an interesting question. I suppose that's a judgement call we can make later, if we decide that `maybe-async` is the basis of the approach we take.